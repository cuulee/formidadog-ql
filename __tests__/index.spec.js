"use strict";

const { graphql } = require("graphql");
const fetch = require("isomorphic-fetch");
require("es6-promise").polyfill();
const { schema } = require("../src/schema");

describe("formidadog-ql server", () => {
  // fetch dogs from endpoint, since our schema initalization requires
  // that they be available at runtime. it's small enough that it shouldn't
  // impact test time too much
  let dogs;
  beforeAll(async () => {
    const res = await fetch("https://rawgit.com/FormidableLabs/dogs/master/dogs.json");
    const data = await res.json();
    dogs = data.map((dog) => ({ ...dog,
      likes: 0 }));
  });

  it("executes the dogs query", async () => {
    const query = `
      query dogs {
        dogs {
          key
          name
          breed
          color
          imageUrl
          description
          likes
        }
      }
    `;

    const result = await graphql(
      schema(dogs),
      query
    );

    expect(result).toMatchSnapshot();
  });

  it("executes the dog query", async () => {
    const query = `
      query dog($key: ID!) {
        dog(key: $key) {
          breed
          name
        }
      }
    `;

    const result = await graphql(
      schema(dogs),
      query,
      undefined,
      undefined,
      {
        key: "VmeRTX7j-"
      }
    );

    expect(result).toMatchSnapshot();
    expect(result.data.dog.name).toEqual("Dixie");
  });

  it("executes the likeDog mutation", async () => {
    const mutation = `
      mutation likeDog($key: ID!) {
        likeDog(key: $key) {
          name
          breed
          likes
        }
      }
    `;

    const result = await graphql(
      schema(dogs),
      mutation,
      undefined,
      undefined,
      {
        key: "VmeRTX7j-"
      }
    );

    expect(result).toMatchSnapshot();
    expect(result.data.likeDog.likes).toEqual(1);
  });

  it("executes the likeAllDogs mutation", async () => {
    const mutation = `
      mutation likeAllDogs {
        likeAllDogs {
          likes
        }
      }
    `;

    const result = await graphql(
      schema(dogs),
      mutation
    );

    expect(result).toMatchSnapshot();
    expect(result.data.likeAllDogs.every((dog) => dog.likes > 0)).toBe(true);
  });
});
