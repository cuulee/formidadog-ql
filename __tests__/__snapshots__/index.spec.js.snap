// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formidadog-ql server executes the dog query 1`] = `
Object {
  "data": Object {
    "dog": Object {
      "breed": "Pit Mix",
      "name": "Dixie",
    },
  },
}
`;

exports[`formidadog-ql server executes the dogs query 1`] = `
Object {
  "data": Object {
    "dogs": Array [
      Object {
        "breed": "Chihuahua",
        "color": "Tan",
        "description": "This sassy little dog has a super-size personality. He knows what he wants and goes after it with single-minded determination. For his size, he’s an excellent watchdog, but he can be yappy if he’s not taught to moderate his barking.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/milo.jpg",
        "key": "r17cRT2uW",
        "likes": 0,
        "name": "Milo",
      },
      Object {
        "breed": "Labrador Retriever",
        "color": "Chocolate",
        "description": "The Labrador Retriever was bred to be both a friendly companion and a useful working dog breed. Historically, he earned his keep as a fisherman’s helper: hauling nets, fetching ropes, and retrieving fish from the chilly North Atlantic.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/beau.jpg",
        "key": "H1jyJC2_b",
        "likes": 0,
        "name": "Beau",
      },
      Object {
        "breed": "Pit Mix",
        "color": "Brown",
        "description": "Pit Bull dogs have historically been known as courageous and heroic animals. Pit Bulls are naturally friendly, active dogs that need a lot of exercise. Most Pits are patient and adore children, which makes them wonderful family dogs.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/dixie.jpg",
        "key": "VmeRTX7j-",
        "likes": 0,
        "name": "Dixie",
      },
      Object {
        "breed": "Goldendoodle",
        "color": "Tan",
        "description": "Goldendoodle is a hybrid of a Golden Retriever and a Standard Poodle. While Labradoodles were bred to be guide dogs for vision-impaired people with dog-sensitive allergies, Goldendoodles have been bred primarily to be family pets, although they can also make terrific service dogs.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/leo.jpg",
        "key": "rJKI1Rhu-",
        "likes": 0,
        "name": "Leo",
      },
      Object {
        "breed": "Mutt with Border Collie DNA",
        "color": "Black",
        "description": "Sophie does tricks, enthusiastic barker, fan of the boss’s organic salmon, diagnosed OCD.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/sophie.jpg",
        "key": "tEQM4Lzj_",
        "likes": 0,
        "name": "Sophie",
      },
      Object {
        "breed": "Chihuahua",
        "color": "Brown",
        "description": "Fourteen pounds of speedy, food-seeking, barkiness.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/rusty.jpg",
        "key": "cLnG8C2d_",
        "likes": 0,
        "name": "Rusty",
      },
      Object {
        "breed": "Beagle",
        "color": "White and Brown",
        "description": "Beagles are scenthounds, meaning they live to use their nose. They’re a comfortable size to tote around in your car, simple to groom, and their exercise needs are easily met with a long, meandering walk that gives them plenty of time to sniff.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/lucy.jpg",
        "key": "Sk1GkChuZ",
        "likes": 0,
        "name": "Lucy",
      },
      Object {
        "breed": "West Highland White Terrier",
        "color": "White",
        "description": "You may recognize the Westie from her long-running stint as the mascot for Cesar pet food, but she’s more than just a cute face. A true terrier, she’s a fast and clever hunter, plus her lighthearted nature makes for a pet who’s always game for some fun.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/callie.jpg",
        "key": "HJMVJR3ub",
        "likes": 0,
        "name": "Callie",
      },
      Object {
        "breed": "Pomeranian",
        "color": "Brown",
        "description": "Tiny and bright-eyed, the spunky Pomeranian greets the world with endless curiosity and a sure sense that he’s the cutest thing around. He is clever, adaptable, and happy, whether hanging at home or performing as a top athlete on an agility course.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/boba.jpg",
        "key": "BknG9C2d-",
        "likes": 0,
        "name": "Boba",
      },
      Object {
        "breed": "Terrier-Schnauzer Mix",
        "color": "Salt & Pepper",
        "description": "Advancing the collected intelligence of the canine species, Cleo is smart, compassionate, and v good at snuggling. Hobbies include curling up like a donut, barking at mail-carriers, and frequent zoomies.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/cleo.jpg",
        "key": "Bkn7Hq9d-",
        "likes": 0,
        "name": "Cleo",
      },
      Object {
        "breed": "Golden Retriever",
        "color": "Yellow",
        "description": "Levi, aka 'Air Spud' is a very large adult boy who is fascinated by tennis balls & jerky treats. His personality often resembles that of a potato, though he is a very happy & big-hearted creature. Currently pursuing swimming & whatever the humans left out on the kitchen counter.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/levi.jpg",
        "key": "uTW2q9d_",
        "likes": 0,
        "name": "Levi",
      },
      Object {
        "breed": "Mutt",
        "color": "Black",
        "description": "Land seal. Tries hard af.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/harley.jpg",
        "key": "1HHChuZd8",
        "likes": 0,
        "name": "Harley",
      },
      Object {
        "breed": "Labrador Retriever",
        "color": "Chocolate",
        "description": "Loves swimming and long walks. Snores like a human.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/laughlan1.jpg",
        "key": "rWD2q1s_",
        "likes": 0,
        "name": "Laughlan",
      },
      Object {
        "breed": "Husky Dane mix",
        "color": "Black",
        "description": "Found on Lookout Mountain in Colorado, Otis is an energetic, noisy mutt that will always choose to dive into cacti in pursuit of lizards. He is currently interested in anything that moves or makes a strange sound.",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/otis.jpg",
        "key": "bQo_598qR",
        "likes": 0,
        "name": "Otis",
      },
      Object {
        "breed": "Labrador Retriever",
        "color": "Black",
        "description": "A larger, fluffier, darker colored version of Laughlan. Very good at cuddling",
        "imageUrl": "https://raw.githubusercontent.com/FormidableLabs/dogs/master/boone.jpg",
        "key": "sQm_623qM",
        "likes": 0,
        "name": "Boone",
      },
    ],
  },
}
`;

exports[`formidadog-ql server executes the likeAllDogs mutation 1`] = `
Object {
  "data": Object {
    "likeAllDogs": Array [
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 2,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
      Object {
        "likes": 1,
      },
    ],
  },
}
`;

exports[`formidadog-ql server executes the likeDog mutation 1`] = `
Object {
  "data": Object {
    "likeDog": Object {
      "breed": "Pit Mix",
      "likes": 1,
      "name": "Dixie",
    },
  },
}
`;
